tests/3d_examples/test_3d_dambreak/src/Dambreak.cpp

```diff
@@ -41,8 +41,9 @@ class WaterBlock : public FluidBody
 		{
 			Vecd halfsize_water(0.5 * LL, 0.5 * LH, 0.5 * LW);
 			Vecd translation_water = halfsize_water;
-			body_shape_ = new ComplexShape(body_name);
-			body_shape_->addBrick(halfsize_water, resolution, translation_water, ShapeBooleanOps::add);
+			ComplexShapeTriangleMesh *mesh = new ComplexShapeTriangleMesh();
+			mesh->addBrick(halfsize_water, resolution, translation_water, ShapeBooleanOps::add);
+			body_shape_ = new ComplexShape(mesh);
 		}
```

Comment:
Looks like this part is complex than previous version, what's the principle for this modification?

Reply:
I have extended the ComplexShape class to a higher level, where it could be polymesh(stl) or image(3d, a level set distance map) . ComplexShapeTriangleMesh handles polymesh, while ComplexShapeImageMesh deals with ImageData (MHD format, 3D for now).